src/main/kotlin/com/vk/admstorm/ssh/SshConnectionService.kt

```diff
@@ -185,7 +177,24 @@ class SshConnectionService(private var myProject: Project) : Disposable {
                             })
                             .show()
 
-                        LOG.warn("Failed to connect", e)
+                        LOG.warn("Failed to connect", ex)
+                    } catch (ex: TimeoutException) {
+                        if (indicator.isCanceled) {
+                            LOG.info("Cancelled by user", ex)
+                            return
+                        }
+
+                        AdmNotification("Don't forget to touch the yubikey if it blinks when using the AdmStorm plugin's features")
+                            .withTitle("Yubikey waiting timeout")
+                            .withActions(AdmNotification.Action("Reconnect...") { _, notification ->
+                                notification.expire()
+                                connectWithConnector(connector, onSuccessful)
+                            }).show()
+
+                        LOG.info("Yubikey waiting timeout", ex)
+                    } catch (ex: Exception) {
+                        LOG.error("Unhandled exception ${ex.javaClass.name}")
```

Comment:
This is good, but in a good way it is necessary to inform the user that something went wrong.
I would just throw the exception further

Reply:
Added notification